src/libostree/ostree-repo.c

```diff
@@ -4691,7 +4687,7 @@ _ostree_repo_gpg_verify_with_metadata (OstreeRepo          *self,
 
       if (!_ostree_gpg_verifier_add_keyring_dir (verifier, self->repodir,
                                                  cancellable, error))
-        goto out;
+        return NULL;
```

Comment:
I'm not going to object to this, but so far in ostree we haven't been using "early returns" very much, mostly because when I was originally writing it the autocleanups weren't pervasive, so I ended up having some manual cleanup in `out:`.

So this is OK, and I'm in favor of slowly porting function by function.